packages/terminal/src/node/shell-terminal-server.ts

```diff
@@ -33,19 +33,38 @@ export class ShellTerminalServer extends BaseTerminalServer {
         super(processManager, logger);
     }
 
-    create(options: IShellTerminalServerOptions): Promise<number> {
+    async create(options: IShellTerminalServerOptions): Promise<number> {
         try {
-            options.env = options.env ? options.env : {};
+            options.env = this.mergeProcessEnv(options.env);
             this.mergedCollection.applyToProcessEnvironment(options.env);
             const term = this.shellFactory(options);
             this.postCreate(term);
-            return Promise.resolve(term.id);
+            return term.id;
         } catch (error) {
             this.logger.error('Error while creating terminal', error);
-            return Promise.resolve(-1);
+            return -1;
         }
     }
 
+    /**
+     * Empty string values will be removed from the final env.
+     *
+     * @param env desired environment to merge with `process.env`.
+     */
+    protected mergeProcessEnv(env: Record<string, string | null> = {}): Record<string, string> {
+        // eslint-disable-next-line no-null/no-null
+        const mergedEnv: Record<string, string> = Object.create(null);
+        for (const [key, value] of Object.entries(process.env)) {
+            // if `env` already has a key it means it overrides the variable from `process.env`
```

Comment:
I find the comments here a little odd. Maybe this?

```suggestion
            // Ignore keys from `process.env` that are overridden in `env`. Accept only non-empty strings.
            if (!(key in env) && value) { mergedEnv[key] = value; }
        }
        for (const [key, value] of Object.entries(env)) {
            // Accept only non-empty strings from the `env` object.
```

Reply:
Done, Thanks !